test/topology_experimental_raft/test_alternator.py

```diff
@@ -318,9 +318,85 @@ async def check_two_live_nodes():
     #    await manager.server_stop(server.server_id)
     await task
 
-# TODO: add a more thorough test for /localnodes, creating a cluster with
-# multiple nodes in multiple data centers, and check that we can get a list
-# of nodes in each data center.
+@pytest.mark.asyncio
+async def test_localnodes_multi_dc_multi_rack(manager: ManagerClient):
+    """A test for /localnosts on a more general setup, with multiple DCs and
+       multiple racks - an 8-node setup with two DCs, two racks in each, and
+       two nodes in each rack.
+       Test both the default of returning the nodes on DC of the server being
+       connected - and the "dc" and "rack" options for explicitly choosing a
+       specific dc and/or rack.
+    """
+    # Start 8 nodes on two different dcs (called "dc1" and "dc2") and two
+    # different rack ("rack1" and "rack2"), two nodes in each rack.
+    config = alternator_config | {
+        'endpoint_snitch': 'GossipingPropertyFileSnitch'
+    }
+    servers = {}
+    for dc in ['dc1', 'dc2']:
+        for rack in ['rack1', 'rack2']:
```

Comment:
`dc1/dc2` & `rack1/rack2` are used many times across this testcase, it's worth to extract these names into variables: not to accidentally make a typo when referring to them and in case we want to rename them, we'll only need to change a single place, e.g. when introducing `rack11` instead of `rack1`, because we'll be adding more racks to the test body - maybe not the best example, but presents the idea.

Reply:
What will I call these variables? dc1 and dc2? Do you see the irony? :-)

Reply:
I don't know how to answer that. The test is to be easy to read by humans. To me, what you are proposing:
```
dc1 = 'dc1'
dc2 = 'dc2'
for dc in [dc1, dc2]:
```
Just adds a level of indirection and more cognitive burden: Every time I as the reader see `dc1` I need to think in my mine - what does this variable contain? Oh, it's just the string `'dc1'` :-)

I honestly don't understand why you prefer it.

Reply:
Also, your question refers to "accidentally making a typo when referring to them and in case we want to rename them". None of these concerns are relevant to a short test, which would fail if we "accidentally make a typo", and there will be no need to ever improve it in the future (you aren't supposed to improve tests - you're supposed to write new ones) or rename parts of it.